core/src/main/java/com/yahoo/ycsb/workloads/RestWorkload.java

```diff
@@ -0,0 +1,401 @@
+package com.yahoo.ycsb.workloads;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.RandomByteIterator;
+import com.yahoo.ycsb.Workload;
+import com.yahoo.ycsb.WorkloadException;
+import com.yahoo.ycsb.generator.ConstantIntegerGenerator;
+import com.yahoo.ycsb.generator.DiscreteGenerator;
+import com.yahoo.ycsb.generator.ExponentialGenerator;
+import com.yahoo.ycsb.generator.HistogramGenerator;
+import com.yahoo.ycsb.generator.HotspotIntegerGenerator;
+import com.yahoo.ycsb.generator.NumberGenerator;
+import com.yahoo.ycsb.generator.UniformIntegerGenerator;
+import com.yahoo.ycsb.generator.ZipfianGenerator;
+
+/**
+ * Typical RESTFul services benchmarking scenario. Represents a set of client
+ * calling REST operations like HTTP DELETE, GET, POST, PUT on a web service.
+ * This scenario is completely different from CoreWorkload which is mainly
+ * designed for databases benchmarking. However due to the huge overlapping
+ * functionality, a major portion of the code in this class is extracted from
+ * {@link CoreWorkload} class.

```

Comment:
Done. Extended CoreWorkload class and override the functionality specific to RestWorkload. Reduced the class size drastically. Only one more scope of code re-usage is in getKeyChooser method in RestWorkload. In CoreWorkload, the code to initialize a keychooser is spreaded throughout the init method. This could be re-used by creating a method getKeyChooser in CoreWorkload, just like RestWorkload. It would touch a lot of places and has the potential to break things. Please suggest if I should make that change.